src/test/java/io/neonbee/test/endpoint/odata/ODataBatchTest.java

```diff
@@ -0,0 +1,168 @@
+        ODataRequest batchRequest = new ODataRequest(TestService1EntityVerticle.TEST_ENTITY_SET_FQN)
+                .setBatch(readRequest);
+
+        assertODataBatch(requestOData(batchRequest), multipartResponse -> {
+            List<MultipartResponse.Part> parts = multipartResponse.getParts();
+            assertThat(parts).hasSize(1);
+
+            MultipartResponse.Part part = parts.get(0);
+            JsonObject acutal = part.getPayload().toJsonObject();
+            assertThat(acutal.getString("KeyPropertyString")).isEqualTo(id);
+
+            testContext.completeNow();
+        }, testContext);
+    }
+
+    @ParameterizedTest(name = "{index}: with key {0}")
+    @MethodSource("withInvalidKeys")
+    @DisplayName("Read single unkown entity via batch request")
+    void testReadUnknownEntity(String id, VertxTestContext testContext) {
+        ODataRequest readRequest = new ODataRequest(TestService1EntityVerticle.TEST_ENTITY_SET_FQN).setKey(id);
+        ODataRequest batchRequest =
+                new ODataRequest(TestService1EntityVerticle.TEST_ENTITY_SET_FQN).setBatch(readRequest);
+
+        assertODataBatch(requestOData(batchRequest), multipartResponse -> {
+            // verify correct amount of response parts
+            List<MultipartResponse.Part> parts = multipartResponse.getParts();
+            assertThat(parts).hasSize(1);
+
+            // verify expected status code of response parts
+            MultipartResponse.Part part = parts.get(0);
+            assertThat(part.getStatusCode()).isEqualTo(404);
+
+            testContext.completeNow();
+        }, testContext);
+    }
+
+    @Test
+    @DisplayName("Test entity filtering")
+    void testFilterOnly(VertxTestContext testContext) {
+        ODataRequest filterRequest = new ODataRequest(TestService1EntityVerticle.TEST_ENTITY_SET_FQN)
+                .setQuery(Map.of("$filter", "KeyPropertyString in ('id-1', 'id-6')"));
+        ODataRequest batchRequest = new ODataRequest(TestService1EntityVerticle.TEST_ENTITY_SET_FQN)
+                .setBatch(filterRequest);
+
+        assertODataBatch(requestOData(batchRequest), multipartResponse -> {
+            // verify correct amount of response parts
+            List<MultipartResponse.Part> parts = multipartResponse.getParts();
+            assertThat(parts).hasSize(1);
+            MultipartResponse.Part part = parts.get(0);
+
+            // verify correct amount of results
+            JsonArray values = part.getPayload().toJsonObject().getJsonArray("value");
+            assertThat(values.size()).isEqualTo(2);
+
+            testContext.completeNow();
+        }, testContext);
+    }
+
+    @ParameterizedTest(name = "{index}: with key {0}")
+    @MethodSource("withValidKeys")
+    @DisplayName("Verify filter and read operations via one batch request return the same results")
+    void testFilterAndRead(String id, VertxTestContext testContext) {
```

Comment:
What exactly is the purpose of this test? You verified already that Read + key and Read + Filter works as expected.

Reply:
The other tests should validate that different types of requests are processed correctly in isolation (one part per batch request).
Whereas this test shall verify that a batch request with more than one single requests is processed correctly (two parts per batch request).

Reply:
As discussed